src/debugging/netcore/NetCoreDebugHelper.ts

```diff
@@ -0,0 +1,259 @@
+                    fsProvider,
+                    ext.context.globalState,
+                    osProvider,
+                    processProvider
+                );
+            }
+
+            return this.vsDbgClient;
+        };
+    }
+
+    public async provideDebugConfigurations(context: DockerDebugScaffoldContext, options?: NetCoreDebugScaffoldingOptions): Promise<DockerDebugConfiguration[]> {
+        options = options || {};
+        options.appProject = options.appProject || await NetCoreTaskHelper.inferAppProject(context.folder); // This method internally checks the user-defined input first
+
+        return [
+            {
+                name: 'Docker .NET Core Launch',
+                type: 'docker-launch',
+                request: 'launch',
+                preLaunchTask: 'docker-run: debug',
+                netCore: {
+                    appProject: unresolveWorkspaceFolderPath(context.folder, options.appProject)
+                }
+            }
+        ];
+    }
+
+    public async resolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<ResolvedDebugConfiguration | undefined> {
+        debugConfiguration.netCore = debugConfiguration.netCore || {};
+        debugConfiguration.netCore.appProject = await NetCoreTaskHelper.inferAppProject(context.folder, debugConfiguration.netCore); // This method internally checks the user-defined input first
+
+        const { configureSsl, containerName, platformOS } = await this.loadExternalInfo(context.folder, debugConfiguration);
+        const appOutput = await this.inferAppOutput(debugConfiguration.netCore);
+        if (context.cancellationToken && context.cancellationToken.isCancellationRequested) {
+            return undefined;
+        }
+
+        await this.acquireDebuggers(platformOS);
+        if (context.cancellationToken && context.cancellationToken.isCancellationRequested) {
+            return undefined;
+        }
+
+        if (configureSsl) {
+            await this.configureSsl(debugConfiguration, appOutput);
+            if (context.cancellationToken && context.cancellationToken.isCancellationRequested) {
+                return undefined;
+            }
+        }
+
+        const additionalProbingPathsArgs = NetCoreDebugHelper.getAdditionalProbingPathsArgs(platformOS);
+
+        const containerAppOutput = NetCoreDebugHelper.getContainerAppOutput(debugConfiguration, appOutput, platformOS);
+
+        const dockerServerReadyAction = await this.inferServerReadyAction(debugConfiguration, containerName, configureSsl);
+
+        return {
+            name: debugConfiguration.name,
+            type: 'coreclr',
+            request: 'launch',
+            program: debugConfiguration.program || 'dotnet',
+            args: debugConfiguration.args || [additionalProbingPathsArgs, containerAppOutput].join(' '),
+            cwd: debugConfiguration.cwd || platformOS === 'Windows' ? 'C:\\app' : '/app',

```

Comment:
>platformOS [](start = 43, length = 10)

intentionally case-sensitive?

Reply:
Yes, VSCode is case-sensitive in launch and task configs.